core/src/main/java/org/opensearch/sql/expression/function/CollectionUDF/MVAppendFunctionImpl.java

```diff
@@ -0,0 +1,160 @@
+    };
+  }
+
+  @Override
+  public UDFOperandMetadata getOperandMetadata() {
+    return null;
+  }
+
+  private static RelDataType determineElementType(
+      SqlOperatorBinding sqlOperatorBinding, RelDataTypeFactory typeFactory) {
+    RelDataType mostGeneralType = null;
+    boolean hasStringType = false;
+    boolean hasNumericType = false;
+
+    for (int i = 0; i < sqlOperatorBinding.getOperandCount(); i++) {
+      RelDataType operandType = getComponentType(sqlOperatorBinding.getOperandType(i));
+
+      if (isStringType(operandType)) {
+        hasStringType = true;
+      } else if (isNumericType(operandType)) {
+        hasNumericType = true;
+      }
+
+      mostGeneralType = updateMostGeneralType(mostGeneralType, operandType, typeFactory);
+    }
+
+    if (hasStringType && hasNumericType) {
+      return typeFactory.createSqlType(SqlTypeName.VARCHAR);
+    }
+
+    return mostGeneralType != null
+        ? mostGeneralType
+        : typeFactory.createSqlType(SqlTypeName.VARCHAR);
+  }
+
+  private static RelDataType getComponentType(RelDataType operandType) {
+    if (!operandType.isStruct() && operandType.getComponentType() != null) {
+      return operandType.getComponentType();
+    }
+    return operandType;
+  }
+
+  private static boolean isStringType(RelDataType type) {
+    SqlTypeName typeName = type.getSqlTypeName();
+    return typeName == SqlTypeName.VARCHAR || typeName == SqlTypeName.CHAR;
+  }
+
+  private static boolean isNumericType(RelDataType type) {
+    return SqlTypeName.NUMERIC_TYPES.contains(type.getSqlTypeName());
+  }
+
+  private static RelDataType updateMostGeneralType(
+      RelDataType current, RelDataType candidate, RelDataTypeFactory typeFactory) {
+    if (current == null) {
+      return candidate;
+    }
+
+    RelDataType leastRestrictive = typeFactory.leastRestrictive(List.of(current, candidate));
+    return leastRestrictive != null ? leastRestrictive : current;
+  }
+
+  public static class MVAppendImplementor implements NotNullImplementor {
+    @Override
+    public Expression implement(
+        RexToLixTranslator translator, RexCall call, List<Expression> translatedOperands) {
+      return Expressions.call(
+          Types.lookupMethod(MVAppendFunctionImpl.class, "mvappend", Object[].class),
+          Expressions.newArrayInit(Object.class, translatedOperands));
+    }
+  }
+
+  public static Object mvappend(Object... args) {
+    List<Object> elements = collectElements(args);
+    return elements.isEmpty() ? null : elements;
+  }
+
+  private static List<Object> collectElements(Object... args) {
+    List<Object> elements = new ArrayList<>();
+
+    for (Object arg : args) {
+      if (arg == null) {
+        continue;
+      }
+
+      if (arg instanceof List) {
+        addListElements((List<?>) arg, elements);
+      } else if (arg.getClass().isArray()) {
+        addArrayElements((Object[]) arg, elements);
+      } else {
```

Comment:
Not sure this code path been used or not. If yes, it will throw exception

class [I cannot be cast to class [Ljava.lang.Object; ([I and [Ljava.lang.Object; are in module java.base of loader 'bootstrap')
java.lang.ClassCastException: class [I cannot be cast to class [Ljava.lang.Object; ([I and [Ljava.lang.Object; are in module java.base of loader 'bootstrap')

for example

```
  @Test
  public void test() {
    Object arg = new int[] {1, 2, 3};
    List<Object> objects = new ArrayList<>();

    if (arg.getClass().isArray()) {
      addArrayElements((Object[]) arg, objects);
    }
  }

  private static void addArrayElements(Object[] array, List<Object> elements) {
    for (Object item : array) {
      if (item != null) {
        elements.add(item);
      }
    }
  }
```